app/_components/PoiCard.tsx

```diff
@@ -144,29 +211,66 @@ export function PoiCard({
         {collect && userCoordinates ? (
           <p className="truncate">{payload.description}</p>
         ) : (
-          <Button
-            id={`${id}`}
-            className="w-full mt-4 rounded-lg"
-            
-            onClick={():void  => {
-              if (user) {
-                if (!handleCheckUserInSearchZone()) {
-                  if (trackingPinContext) {
-                    trackingPinContext.setTrackingPin(payload);
-                    setShowPopup && setShowPopup(false);
+          <div>
+            <Button
+              id={`${id}`}
+              className="w-full mt-4 rounded-lg"
+              onClick={(): void => {
+                if (user) {
+                  if (!handleCheckUserInSearchZone()) {
+                    if (trackingPinContext) {
+                      trackingPinContext.setTrackingPin(payload);
+                      setShowPopup && setShowPopup(false);
+                    }
```

Comment:
There's a lot of nested `if` statements here. Is it possible to streamline this?

Reply:
extracted one "if" statement